### ISSUE TITLE
'update_notification_last' is causing a mess

### ISSUE THREAD

**Issue by Naheel-Azawy** [SamSchott/maestral#108]
**Describe the bug**
I installed a while ago, got this error, and uninstalled. I tried again today and still the same. Something weird related to `update_notification_last` is happening..

```
~> maestral
Traceback (most recent call last):
  File "/usr/lib/python3.8/configparser.py", line 789, in get
    value = d[option]
  File "/usr/lib/python3.8/collections/__init__.py", line 891, in __getitem__
    return self.__missing__(key)            # support subclasses that define __missing__
  File "/usr/lib/python3.8/collections/__init__.py", line 883, in __missing__
    raise KeyError(key)
KeyError: 'update_notification_last'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/maestral", line 11, in <module>
    load_entry_point('maestral==0.6.4', 'console_scripts', 'maestral')()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2859, in load_entry_point
    return ep.load()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2450, in load
    return self.resolve()
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2456, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3.8/site-packages/maestral/cli.py", line 28, in <module>
    run_housekeeping()
  File "/usr/lib/python3.8/site-packages/maestral/utils/housekeeping.py", line 134, in run_housekeeping
    migrate_user_config(config_name)
  File "/usr/lib/python3.8/site-packages/maestral/utils/housekeeping.py", line 57, in migrate_user_config
    update_notification_last = old_conf.get('app', 'update_notification_last')
  File "/usr/lib/python3.8/configparser.py", line 792, in get
    raise NoOptionError(option, section)
configparser.NoOptionError: No option 'update_notification_last' in section: 'app'
```

**To Reproduce**
Just run it...

**Expected behaviour**
It works...

**System (please complete the following information):**
 - Maestral version: 0.6.4
 - OS: 5.5.11-arch1-1
 - Desktop environment: non, here's my [dot files](https://github.com/Naheel-Azawy/naheel-dotfiles) if needed
 - Qt version: PyQt5-5.14.2

**Additional context**
Nice job! I'm really interested!

**Comment by SamSchott** [comment]
Sorry to hear about this. This looks like the config file was not properly migrated in one of the previous updates, likely from v0.6.1. There have issues with the migration in some cases unfortunately.

The easiest way to fix this is to delete the current config file at `~/.config/maestral/maestral.ini` and try to start maestral again. This will prompt you to re-link your Dropbox account. You can then choose to keep / merge you local Dropbox folder which will trigger a reindexing but no unneeded downloads.

Alternatively, you could post the config file (with personal information redacted) and I can see if it can be fixed manually. This way, no relinking would be necessary...

**Comment by Naheel-Azawy** [comment]
```
~> rm -rf ~/.config/maestral
~> maestral link
No Dropbox account linked.
1. Go to: https://www.dropbox.com/oauth2/authorize?...
2. Click "Allow" (you might have to log in first).
3. Copy the authorization token.
Enter the authorization token here: <TOKEN>
Could not access the user keyring to save your authentication token. Please make sure that the keyring is unlocked.
```

I guess this means deleting the config solved it but I'm not sure what keyring is it talking about?..

**Comment by SamSchott** [comment]
Hm, this is unusual. This message will appear if you have Gnome Keyring, KWallet (in case of KDE) or any other password storage that implements the SecretService dbus interface installed. Maestral will then try to store your auth token in this keyring. But, as hinted above, such keyrings typically come with desktop environments and are automatically decrypted / unlocked on login.

You can check which keyring Maestral is using as follows:

```Python
from maestral.utils.backend import get_keyring_backend

ring = get_keyring_backend('maestral')
print(ring)
```

You will then need to ensure that it is unlocked before starting Maestral. 

Alternatively, you can force Maestral to store the Dropbox auth token in a plain text file by changing the `keyring` option from `automatic` to `keyrings.alt.file.PlaintextKeyring`.

**Comment by Naheel-Azawy** [comment]
Sorry for my slow response...
```python
>>> from maestral.utils.backend import get_keyring_backend
>>> ring = get_keyring_backend('maestral')
>>> print(ring)
keyring.backends.SecretService.Keyring (priority: 5)
```

I did set the backend to `keyrings.alt.file.PlaintextKeyring` and:
```
~> maestral link
No Dropbox account linked.
1. Go to: https://www.dropbox.com/oauth2/authorize?...
2. Click "Allow" (you might have to log in first).
3. Copy the authorization token.
Enter the authorization token here: ...
 > Credentials written.
 > Warning: No supported keyring found, Dropbox credentials stored in plain text.
```

I know this is not safe but looked good so far! However...

```
~> maestral start
Traceback (most recent call last):
  File "/usr/bin/maestral", line 11, in <module>
    load_entry_point('maestral==0.6.4', 'console_scripts', 'maestral')()
  File "/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/maestral/cli.py", line 134, in wrapper
    return func(*args, **kwargs)
  File "/usr/lib/python3.8/site-packages/maestral/cli.py", line 342, in start
    m = Maestral(config_name, run=False)
  File "/usr/lib/python3.8/site-packages/maestral/main.py", line 220, in __init__
    self._setup_logging()
  File "/usr/lib/python3.8/site-packages/maestral/main.py", line 295, in _setup_logging
    self.log_handler_journal = journal.JournalHandler(
AttributeError: 'NoneType' object has no attribute 'JournalHandler'
```

**Comment by SamSchott** [comment]
Apologies, you have run into another issue which has been already fixed in the development branch. Maestral will log to the systemd journal if has been started as as a systemd service (or thinks that it has...). The required package `python-systemd` is not a mandatory dependency since it can be difficult to install on some systems. However, the check if `systemd-python` is available was missing.

For now, you can either install [python-systemd](https://www.archlinux.org/packages/extra/x86_64/python-systemd/) or make sure that the environment variable INVOCATION_ID is not set (this is set by systemd for the service process).

**Comment by Naheel-Azawy** [comment]
I installed python-systemd and it worked finally!!! Kind of..

```
~> # First time, it took maybe like 5 minutes
~> maestral start
Please give Dropbox folder location [/home/naheel/Dropbox (Maestral)]:
Would you like to exclude any folders from syncing? [y/N]: y
Please choose which top-level folders to exclude. You can exclude
individual files or subfolders later with "maestral excluded add".
Exclude "/orgmode" from sync? [y/N]: n
Exclude "/others" from sync? [y/N]: y
Exclude "/Tom" from sync? [y/N]: y
Starting Maestral...        [FAILED]
~>
~> maestral start
Starting Maestral...        [FAILED]
~>
~> # Got disappointed...
~> # Then I thought... Let's try the GUI maybe some magic happen...
~> # And it did!!! Even though it took pretty long time but it did sync!!!
~>
~> maestral gui
QApplication: invalid style override 'adwaita-dark' passed, ignoring it.
        Available styles: Windows, Fusion
~>
~> maestral start
Maestral daemon is already running.
~> # Yay!!
```

So, great that it worked and I guess I will start using it regardless! I really needed this.

But...  
1. I guess there's something wrong with running it without GUI? Why does it fail?
2. Why does it take long time (around 2 - 5 minutes)? Is this normal?
3. Regarding the auth token, where is it stored exactly in plain? Maybe I can at least throw it under encfs I guess..

Sorry I'm so unlucky with your software xD

**Comment by SamSchott** [comment]
> I guess there's something wrong with running it without GUI? Why does it fail?

I cannot reproduce this failure but maybe the logs will help. In any case, startup procedures will be different in the next release and the issue might just go away.

 > Why does it take long time (around 2 - 5 minutes)? Is this normal?

This is not normal. I suspect its because Qt is waiting for something over dbus until it eventually times out. Are there any errors printed by Qt?

> Regarding the auth token, where is it stored exactly in plain? Maybe I can at least throw it under encfs I guess..

It's stored at `~/.local/share/python_keyring/keyring_pass.cfg` and yes, encfs would be the best approach.

> Sorry I'm so unlucky with your software xD

I have run into a whole range of unforeseeable platform integration issues with Linux, it really is a mess to support compared to macOS...

**Comment by Naheel-Azawy** [comment]
> I cannot reproduce this failure but maybe the logs will help. In any case, startup procedures will be different in the next release and the issue might just go away.

Sweet! Let's hope for the best. But anyway, I just tried again and it magically started working properly... Both GUI and no GUI. I haven't done any change in my system I guess...

> This is not normal. I suspect its because Qt is waiting for something over dbus until it eventually times out. Are there any errors printed by Qt?

Yeah as shown above:
```
QApplication: invalid style override 'adwaita-dark' passed, ignoring it.
        Available styles: Windows, Fusion
```

But I don't think it is the reason. The GUI itself doesn't take time, it's the starting that does. But anyway, now it's quick too magically... No idea why...

> It's stored at ~/.local/share/python_keyring/keyring_pass.cfg and yes, encfs would be the best approach.

Thanks!

> I have run into a whole range of unforeseeable platform integration issues with Linux, it really is a mess to support compared to macOS...

Dockerize maybe?

**Comment by SamSchott** [comment]
Closing since the original issue has been resolved and the migration code will be removed in the next release.

Feel free to open a new issue if the startup failure happens again.
